src/app/server/CommissioningWindowManager.cpp

```diff
@@ -313,6 +313,11 @@ CHIP_ERROR CommissioningWindowManager::StopAdvertisement()
     DeviceLayer::ConnectivityMgr().RequestSEDFastPollingMode(false);
 #endif
 
+    // Stop advertising commissioning mode, since we're not accepting PASE
+    // connections right now.  If we start accepting them again (via
+    // OpenCommissioningWindow) that will call StartAdvertisement as needed.
+    app::DnssdServer::Instance().StartServer(Dnssd::CommissioningMode::kDisabled);
```

Comment:
It's odd that `StartServer` is the name for a "stop". I would have expected `app::DnssdServer::Instance().SetCommissionableDiscoveryMode(Dnssd::CommissioningMode::kDisabled)`.

Reply:
It's more of a "RestartServer" in practice....  Renaming the API and cleaning it up is worthwhile, but I would argue out of scope for this PR.

Reply:
Is this line the essence of the fix?

Reply:
Yes, this line is the essence of the fix.  The bits in Dnssd.h/.cpp are just to make this line work, and the other changes in this file are just to make StartAdvertising/StopAdvertising more symmetric in terms of handling all the advertising, instead of handling BLE in StartAdvertising/StopAdvertising but manually doing DNS-SD bits outside it.